source/val/validate_builtins.cpp

```diff
@@ -2088,8 +2088,11 @@ spv_result_t BuiltInsValidator::ValidateVertexIndexAtDefinition(
 }
 
 spv_result_t BuiltInsValidator::ValidateVertexIdOrInstanceIdAtDefinition(
-    const Instruction& inst) {
-  if (spvIsVulkanEnv(_.context()->target_env)) {
+    const Decoration& decoration, const Instruction& inst) {
+  const SpvBuiltIn label = SpvBuiltIn(decoration.params()[0]);
+  bool allow_instance_id = _.HasCapability(SpvCapabilityRayTracingNV) &&
+                           label == SpvBuiltInInstanceId;
```

Comment:
I'm still not convinced that this is an adequate check.
From surveying other validation examples in this file, it seems like we'd need to add an "AtReference" version of the check and add in something like 

```
for (const SpvExecutionModel execution_model : execution_models_) {
    switch (execution_model) {
        case SpvExecutionModelIntersectionNV:
        case SpvExecutionModelAnyHitNV:
        case SpvExecutionModelClosestHitNV:
            break; // allowed
       default: 
           // error...
    }
} 
```

And that's where I got bogged down when I investigated adding the change :-/

Ie - having RayTracingNV capability should NOT enable this in existing stages, and should NOT enable it in all new stages that area added under RayTracingNV capability either.

Reply:
My goal is not to validate the Nvidia extension. I'm just trying to avoid failing a shader that should be valid.

Reply:
We can file a followup issue for the more precise check.

Reply:
> My goal is not to validate the Nvidia extension. I'm just trying to avoid failing a shader that should be valid.

Those seem to be contradictory. 

> We can file a followup issue for the more precise check.

Sure fine.

Reply:
> Those seem to be contradictory.

They are not. False positives from the validator (invalid module being found valid) are more acceptable than false negatives. This change will eliminate the false negatives, but introduce false positives. I'm ok with that.